zndraw/server/events.py

```diff
@@ -340,21 +365,43 @@ def scene_pause(data):
 @io.on("modifier:register")
 def modifier_register(data):
     data["token"] = session["token"]
+    if "modifier" not in app.config["PER-TOKEN-DATA"][session["token"]]:
+        # this is a dict for the pyclient, but it has to be for every webclient ...
+        app.config["PER-TOKEN-DATA"][session["token"]]["modifier"] = {}
 
     try:
         # we can only register one modifier at a time
         name = data["modifiers"][0]["name"]
-        if name in app.config["MODIFIER"]:
-            # issue with the same modifier name on different webclients / tokens!
-            # only for default we need to ensure, there is only one.
-            raise ValueError(f"Modifier {name} is already registered.")
-        app.config["MODIFIER"][name] = request.sid
+        if name in app.config["MODIFIER"]["default_schema"]:
+            msg = f"Modifier {name} is already registered (default)."
+            log.critical(msg)
+            emit("message:log", msg, to=request.sid)
+
+        if name in app.config["PER-TOKEN-DATA"][session["token"]]["modifier"]:
+            msg = f"Modifier {name} is already registered."
+            log.critical(msg)
+            emit("message:log", msg, to=request.sid)
+        # get the key from the value request.sid by inverting the dict
+        _tmp = {v: k for k, v in app.config["pyclients"].items()}
+        app.config["PER-TOKEN-DATA"][session["token"]]["modifier"][name] = _tmp[
+            request.sid
+        ]
+        log.critical(f'{app.config["PER-TOKEN-DATA"] = }')
+        log.critical(f'{app.config["pyclients"] = }')
+
         if data["modifiers"][0]["default"]:
+            if not session["authenticated"]:
+                msg = "Unauthenticated users cannot register default modifiers."
+                log.critical(msg)
+                emit("message:log", msg, to=request.sid)
+            log.critical(app.config["MODIFIER"])
+            log.critical(f"{name = }")
```

Comment:
Should probably have more informative logging here.

Reply:
These will be removed.